### PR TITLE
Move the Icons demo into a card, and refactor it to use a Table widget.

### PR DESCRIPTION
**Pull request by amirh**
This is in preparation for adding a second card with a demo for animated
icons.

Table refactor is just a cleanup (that also fixes an alignment bug).

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/material/icons_demo.dart [hunk 1]
```diff
@@ -44,7 +44,34 @@ class IconsDemoState extends State<IconsDemo> {
...
+      ),
+    );
+  }
+}
+
+class _IconsDemoCard extends StatelessWidget {
+
+  final VoidCallback handleIconButtonPress;
+
+  const _IconsDemoCard(this.handleIconButtonPress);
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/material/icons_demo.dart]: always put the constructor first, that way we can tell at a glance if there's any constructors or if there's an implicit constructor instead.

File: examples/flutter_gallery/lib/demo/material/icons_demo.dart [hunk 2]
```diff
@@ -53,72 +80,45 @@ class IconsDemoState extends State<IconsDemo> {
...
-    return new SizedBox(
-      height: size.toDouble() + 16.0, // to match an IconButton's padded height
-      child: new Center(
-        child: new Text('$size', style: style)
-      )
+  Widget _centeredText(String label) =>
+    new Padding(
+      // Match the default padding of IconButton.
+      padding: const EdgeInsets.all(8.0),
+      child: new Text(label, textAlign: TextAlign.center)
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/material/icons_demo.dart]: nit: trailing commas

**Comment by Hixie** [comment]
LGTM

**Comment by amirh** [comment]
addressed comments

**Pull request merged**